master/internal/core_experiment.go

```diff
@@ -352,10 +352,79 @@ type CreateExperimentParams struct {
 	ValidateOnly  bool            `json:"validate_only"`
 }
 
+// getResourcePoolTaskContainerDefaults is a helper function that partially parses an experiment
+// config, and returns a TaskContainerDefaultsConfig that can be fed to
+// model.DefaultExperimentConfig().  This is of course way too convoluted to be "nice", but all of
+// this will go away when we can use schemas.Merge() rather than layers of json.Unmarshal().
+func getResourcePoolTaskContainerDefaults(
```

Comment:
This behavior is inconsistent with how per-resource-pool settings are done elsewhere (e.g. scheduler). In that case we use the global defaults if there are no resource-pool specific settings otherwise we use only the per-resource-pool settings (no partial inheritance). https://docs.determined.ai/latest/topic-guides/system-concepts/resource-pool.html#migrating-to-resource-pools

I think having different inheritance rules depending on which field we are talking about will be confusing.

Reply:
After thinking about it, I agree that the merging behavior is the wrong behavior.

Also I think it goes without saying that we should be matching the other resource pool behaviors (which I was unaware of, so nice catch).